index.js

```diff
@@ -33,26 +37,111 @@ CredentialsAuthenticator.prototype.authenticate = function authenticate(plexApi,
-        }
-        if (res.statusCode !== 201) {
-            return callback(new Error('Invalid status code in authentication response from Plex.tv, expected 201 but got ' + res.statusCode));
-        }
+            if (!token) {
+                throw new Error('Couldnt not find authentication token in response from Plex.tv :(');
+            }
 
-        token = extractAuthToken(xmlBody);
-        if (!token) {
-            return callback(new Error('Couldnt not find authentication token in response from Plex.tv :('));
-        }
+            self.emit('authToken', token);
 
-        self.emit('token', token);
-        callback(null, token);
-    });
+            return token;
+        })
+        .catch(errors.StatusCodeError, err => {
+            throw new Error('Invalid status code in authentication response from Plex.tv, expected 201 but got ' + err.statusCode);
+        })
+        .catch(errors.RequestError, err => {
+            throw new Error('Error while requesting https://plex.tv for authentication: ' + String(err));
+        })
+        .then(token => {
+            if (self.homeUser && self.homeUser.name) {
+                let plexContext = {
+                    plexApi : plexApi,
+                    token   : token
+                };
+
+                return getHomeUser(plexContext)
+                    .then(xml  => findUserByName(xml, self.homeUser.name))
+                    .then(user => switchUser(plexContext, user, self.homeUser.pin))
+                    .then(extractAuthToken)
+                    .then(token => getAccessToken({token : token, plexApi : plexApi}))
+                    .then(accessToken => {
+                        self.emit('accessToken', accessToken);
+                        self.emit('token', accessToken);
+                        return token;
+                    });
+            }
+            self.emit('token', token);
+            return token;
+        })
+        .asCallback(callback);
 };
 
+function getHomeUser(plexContext) {
+    return request.get(createRequestOpts(plexContext, 'https://plex.tv/api/home/users'));
+}
+
+function createRequestOpts(plexContext, url) {
+    return {
+        url     : url,
+        headers : headers(plexContext.plexApi, { 'X-Plex-Token': plexContext.token })
+    };
+}
+
+function getAccessToken(plexContext, callback) {
+    return request.get(createRequestOpts(plexContext, 'https://plex.tv/api/resources?includeHttps=1'))
+        .then(parseString)
+        .then(extractAccessToken);
+}
+
+function extractAccessToken(result) {
+    let token = null;
+    let hasDevices = result.MediaContainer.Device && result.MediaContainer.Device.some(device => {
+        token = device.$.accessToken;
+        return true;
+    });
+
+    if (hasDevices) {
+        return token;
+    }
+
+    throw new Error('No devices available to extractAccessToken from');
```

Comment:
I got this error when creating a managed user with *no access* to any of the libraries on the Plex Server. Maybe this error message could be more descriptive about that?

Something like `Couldn't find any devices this managed user has access to`